tests/proxy/ringbuffer/ringbuffer_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ringbuffer
+
+import (
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/core"
+	"github.com/hazelcast/hazelcast-go-client/internal/common"
+	. "github.com/hazelcast/hazelcast-go-client/rc"
+	. "github.com/hazelcast/hazelcast-go-client/tests"
+	"log"
+	"testing"
+)
+
+var ringbuffer core.Ringbuffer
+var client hazelcast.IHazelcastInstance
+
+const capacity int64 = 10
+const ringbufferName = "ClientRingbufferTestWithTTL"
+
+func TestMain(m *testing.M) {
+	remoteController, err := NewRemoteControllerClient("localhost:9701")
+	if remoteController == nil || err != nil {
+		log.Fatal("create remote controller failed:", err)
+	}
+	cluster, err := remoteController.CreateCluster("3.9", DefaultServerConfig)
+	remoteController.StartMember(cluster.ID)
+	client, _ = hazelcast.NewHazelcastClient()
+	ringbuffer, _ = client.GetRingbuffer(ringbufferName)
+	m.Run()
+	client.Shutdown()
+	remoteController.ShutdownCluster(cluster.ID)
+}
+
+func TestRingbufferProxy_Name(t *testing.T) {
+	if ringbufferName != ringbuffer.Name() {
+		t.Errorf("Name() failed")
+	}
+}
+
+func TestRingbufferProxy_ServiceName(t *testing.T) {
+	serviceName := common.SERVICE_NAME_RINGBUFFER_SERVICE
+	if serviceName != ringbuffer.ServiceName() {
+		t.Errorf("ServiceName() failed")
+	}
+}
+
+func TestRingbufferProxy_PartitionKey(t *testing.T) {
+	if ringbufferName != ringbuffer.PartitionKey() {
+		t.Errorf("PartitionKey() failed")
+	}
+}
+
+func fillRingbuffer(capacity int64) {
+	for i := int64(0); i < capacity; i++ {
+		ringbuffer.Add(i, core.OverflowPolicyOverwrite)
+	}
+}
+func destroyAndCreate() {
+	ringbuffer.Destroy()
+	ringbuffer, _ = client.GetRingbuffer(ringbufferName)
+}
+
+func TestRingbufferProxy_Capacity(t *testing.T) {
+	defer destroyAndCreate()
+	fillRingbuffer(capacity)
+	cap, err := ringbuffer.Capacity()
+	AssertEqualf(t, err, cap, capacity, "ringbuffer Capacity() failed")
+}
+
+func TestRingbufferProxy_AddSize(t *testing.T) {
```

Comment:
this is not a good test. checking multiple things at once. I would split size and add tests.